thoth/storages/graph/postgres.py

```diff
@@ -3574,10 +3711,21 @@ def _create_python_package_version(
 
         return python_package_version
 
+    @staticmethod
+    def _create_hash(inputs: List[int]) -> str:
+        """Create hash string using ssdeep.
+
+        Reference: https://python-ssdeep.readthedocs.io/en/latest/installation.html#id11
+        """
+        file_ids = "-".join([str(ppr) for ppr in inputs])
```

Comment:
As hash operates on the binary data, it might make sense to keep items as integers rather than converting them to strings.

Reply:
but `ssdeep.hash()` expects a string not a list, how do we pass integers there?

Moreover using only integers sorted might give us the following problem:

- software stack with id 1 , 3 , 4 would results in 134
- software stack with id 1 , 34 would results in 134

and this is not correct, that is why I added the `-` when creating the string, so we don't fall into this bug.

Reply:
Or we could hash all ids and merge them in a single string, but it would be the same I guess, just more expensive

Reply:
> but `ssdeep.hash()` expects a string not a list of int or int, how do we pass integers there?

Hashes work on binary data.

> Moreover using only integers sorted might give us the following problem:
> 
> * software stack with id 1 , 3 , 4 would results in 134
> * software stack with id 1 , 34 would results in 134
> and this is not correct, that is why I added the `-` when creating the string, so we don't fall into this bug.

The issue you described is occurring in the string implementation. As hashes work on binary data, having allocated a fixed size for integers representing packages will yield expected results. The hash itself has no notion about `-` delimiter and the comparison will semantically not work. See comparison using [1].

[1] https://python-ssdeep.readthedocs.io/en/latest/api.html#ssdeep.compare

Reply:
Thanks @fridex 

I tested with

```
import ssdeep
INPUTS_1 = [1, 3, 4]
INPUTS_2 = [1, 34]

hashes = []
for file_ in [INPUTS_1, INPUTS_2]:

    hash_ = ssdeep.hash(bytes(file_))
    hashes.append([file_, hash_])

print(hashes)
print(ssdeep.compare(hashes[0][1], hashes[1][1]))
```

and the result was:
```
[[[1, 3, 4], '3:5:5'], [[1, 34], '3:On:On']]
0
```

So we will pass a list of integers in bytes correct?

Reply:
that does not work for number above 256 :/ what about this libarary https://pypi.org/project/hashids/ @fridex @saisankargochhayat ?

Reply:
Seems popular and might work well, but it's not a fuzzy hash if that's something we require.

Reply:
> Seems popular and might work well, but it's not a fuzzy hash if that's something we require.

Right :/

Reply:
hashids allow to pass different ids actually: `hashids.encode(*table_ids)`